modules/subworkflows/build_indices.nf

```diff
@@ -7,59 +7,74 @@
 // And then initialize channels based on params or indexes that were just built
 
 include { BUILD_INTERVALS } from '../local/build_intervals.nf'
-include { BWAMEM2_INDEX as BWAMEM2_INDEX } from '../nf-core/bwamem2_index.nf'
-include { GATK_CREATE_SEQUENCE_DICTIONARY as GATK_CREATE_SEQUENCE_DICTIONARY } from '../local/gatk_dict.nf'
+include { BWAMEM2_INDEX } from '../nf-core/bwamem2_index.nf'
+include { GATK_CREATE_SEQUENCE_DICTIONARY } from '../local/gatk_dict.nf'
 include {
     HTSLIB_TABIX as HTSLIB_TABIX_DBSNP;
     HTSLIB_TABIX as HTSLIB_TABIX_GERMLINE_RESOURCE;
     HTSLIB_TABIX as HTSLIB_TABIX_KNOWN_INDELS;
     HTSLIB_TABIX as HTSLIB_TABIX_PON;
 } from '../nf-core/htslib_tabix'
-include { SAMTOOLS_FAIDX as SAMTOOLS_FAIDX } from '../nf-core/samtools_faidx.nf'
+include { SAMTOOLS_FAIDX } from '../nf-core/samtools_faidx.nf'
 
 workflow BUILD_INDICES{
     take:
-        ch_dbsnp
-        ch_fasta
-        ch_germline_resource
-        ch_known_indels
-        ch_pon
+        dbsnp
+        fasta
+        germline_resource
+        known_indels
+        pon
         step
 
     main:
 
-    if (!(params.bwa) && params.fasta && 'mapping' in step) 
-        BWAMEM2_INDEX(ch_fasta)
+    if (!(params.bwa) && params.fasta && 'mapping' in step)
+        result_bwa = BWAMEM2_INDEX(fasta)
+    else
+        result_bwa = Channel.empty()
```

Comment:
Glad it works like this, but unhappy with the optics

Reply:
We'll improve it at some point